autoPyTorch/evaluation/train_evaluator.py

```diff
@@ -489,8 +489,12 @@ def eval_train_function(
             with a single instance, being the provided X_train, y_train of a single dataset.
             This instance is a compatibility argument for SMAC, that is capable of working
             with multiple datasets at the same time.
+        evaluator_class (Type[AbstractEvaluator]):
+            the class name of evaluator, when not specified, it is set as vanilla TrainEvaluator
+        evaluator_kwargs: Any
+            additionally evaluation kwargs
     """
-    evaluator = TrainEvaluator(
+    evaluator = evaluator_class(
```

Comment:
We need this argument to tell the function which evaluator to use. I think it is fine to keep it here. Otherwise, I will create a new function with nearly the same arguments.

Reply:
Actually, I would prefer it if we have a new function for it.